### ISSUE TITLE
update NucleiClassification cli

### ISSUE THREAD

**Issue by subinkitware** [DigitalSlideArchive/HistomicsTK#1040]
Tasks need to be done in this issue

- [x] Create a tutorial file for nuclei classification and save it into the docs with sample data
- [x] Create a tutorial file for creating a random forest classifier from the data.
- [x] Update the `NucleiClassification` code to run samples
- [x] Add nuclei annotation and feature extraction code to the cli
- [x] If nuclei annotation and feature files are not given (run feature extraction and annotation by default)
- [x] Create a pickle file with training data from the grand-challenge
- [x] Add the pickle file to the `datastore`

**Comment by cooperlab** [comment]
If using the data from https://nucls.grand-challenge.org/, let's make sure to mention this is a breast cancer classifier. I can help identify input images if needed.

**Comment by subinkitware** [comment]
As a first step, I've done the following using the DSA to create a nuclei classifier.

1. Generated `nucleiFeatures.csv` and `nucleiAnnotation.json` files from a sample WSI with an ROI
2. Since there are no labels in the `nucleiFeatures.csv` I've used K-means clustering to create labels and added that to the csv file to the last column.
3. Trained a `Random forest classifier` and the model is pickled to a `.pkl file`
4. All of this data is loaded into the NucleiClassifier and it was able to generate the nuclei classification in the UI.
5. Some of the functions dealing with the pickled model and data frame were depreciated so I have updated that. 

I was unable to attach the repo as a pull request since the code was not coming from the HistocsTK repo. Here is the sorce containing all the files and code. - https://github.com/subinkitware/NucleiClassifier
**The repo contains the following files.**
- `clustering_data.ipynb` - does K-means clustering and generate labels for the input nuclei features file
- `random_forest_classifier.ipynb` - Generates a scikit-learn random forest model by training the data above.
- `NucleiFeatures.csv` - input feature file
- `OutputNucleiAnot.json` - input nuclei annotation file.

**Comment by subinkitware** [comment]
Hi @cooperlab, It would be great if you can help me select the images from  https://nucls.grand-challenge.org/. I have got access to the dataset. It has a lot of samples with different varieties of segmentations and classes. I liked the samples with just one class of nuclei in them (for example:- all nuclei in the given RGB image are tagged as a tumor). That will make it easier for me to extract the nuclei features.

**Comment by cooperlab** [comment]
@subinkitware we can have a Meet if you want to discuss. I think limiting to pure class tiles might bias things significantly. I realize mapping from detection results to ground truth isn't trivial but I think we can use a heuristic approximation that is simple and lightweight.

**Comment by subinkitware** [comment]
> @subinkitware we can have a Meet if you want to discuss. I think limiting to pure class tiles might bias things significantly. I realize mapping from detection results to ground truth isn't trivial but I think we can use a heuristic approximation that is simple and lightweight.

That would be great. I'm not sure if both of us can see each other on our calendars. I'm available today till 5:00, Tomorrow(9:30-3:00), and Next week Tuesday(12:30 - 5:00), Wed(9:30-4:30), Thursday(""), Friday("").
